boa/src/object/gcobject.rs

```diff
@@ -63,12 +60,32 @@ enum FunctionBody {
 }
 
 impl JsObject {
-    /// Create a new `GcObject` from a `Object`.
+    /// Create a new `JsObject` from an internal `Object`.
     #[inline]
-    pub fn new(object: Object) -> Self {
+    fn from_object(object: Object) -> Self {
         Self(Gc::new(GcCell::new(object)))
     }
 
+    /// Create a new empty `JsObject`, with `prototype` set to `JsValue::Null`
+    /// and `data` set to `ObjectData::ordinary`
+    pub fn empty() -> Self {
+        Self::from_object(Object::default())
+    }
+
+    /// The more general form of `OrdinaryObjectCreate` and `MakeBasicObject`.
+    ///
+    /// Create a `JsObject` and automatically set its internal methods and
+    /// internal slots from the `data` provided.
+    #[inline]
+    pub fn from_proto_and_data(prototype: Option<JsObject>, data: ObjectData) -> Self {
```

Comment:
I would use `Into<Option<JsObject>>` here, so that you can pass a `JsObject` without the `Some`.